extensions/amp-ad-template/0.1/amp-ad-template.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {urls} from '../../../src/config';
+import {Services} from '../../../src/services';
+import {dev} from '../../../src/log';
+import {getMode} from '../../../src/mode';
+import {getAmpdoc} from '../../../src/service';
+import {Bind} from '../../amp-bind/0.1/bind-impl';
```

Comment:
this will compile the whole amp-bind extension into this extension

Reply:
How can I avoid doing so?

Reply:
You might want to get the binder from service.

@choumx correct me if I'm wrong
try: `adoptServiceForEmbedIfEmbeddable()`

Reply:
Don't think this is the way. I get the following error:

`Error: Service amp-bind not found on parent or doesn't implement EmbeddableService.`

Even if this did work. I'd still need some way to call `setState()`.

Reply:
You should retrieve services from the corresponding function in src/services.js --
 in this case,`bindForDocOrNull`